### DISCUSSION TITLE
How do I change the user in a workspace from coder to ${data.coder_workspace.me.owner}

### DISCUSSION THREAD

**Discussion by simonjcarr** [coder/coder#10746] [General] [answered]
How do I change the user and hostname in a workspace. By default it looks something like this `coder@coder-simon-bash-5667c49959-nhb2r ` I would like it to be like this `simon@coder-soxprox`

If I can't change the hostname, I would settle for changing just the username.

I have tried to add a new user in the template startup script. Although the user is created, I can't seem to change to it.

```
sudo adduser --disabled-password --gecos "" ${data.coder_workspace.me.owner}
sudo su ${data.coder_workspace.me.owner}
```

**Comment by code-asher** [comment, marked as answer]
Just FYI, I transferred your discussion from `coder/code-server` to `coder/coder`.

This might help:

https://github.com/coder/coder/blob/ba955f44d04dbf976af32ffd6eb274c4e602ee0c/dogfood/main.tf#L275-L276

Not sure about the user, though.  Maybe check out this template as an example:

https://github.com/coder/coder/blob/ba955f44d04dbf976af32ffd6eb274c4e602ee0c/examples/templates/docker/main.tf
